sdk/tests/conformance2/context/context-attributes-depth-stencil-antialias-obeyed.html

```diff
@@ -0,0 +1,296 @@
+{
+    var colorOffset = vertices.byteLength;
+
+    var vbo = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, vbo);
+    gl.bufferData(gl.ARRAY_BUFFER, colorOffset + colors.byteLength, gl.STATIC_DRAW);
+    gl.bufferSubData(gl.ARRAY_BUFFER, 0, vertices);
+    gl.bufferSubData(gl.ARRAY_BUFFER, colorOffset, colors);
+
+    gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+    gl.enableVertexAttribArray(0);
+    gl.vertexAttribPointer(1, 4, gl.UNSIGNED_BYTE, true, 0, colorOffset);
+    gl.enableVertexAttribArray(1);
+
+    gl.drawArrays(gl.TRIANGLES, 0, vertices.length / 3);
+}
+
+function testDefault()
+{
+    debug("");
+    debug("Testing default attributes: { antialias: true, depth: true, stencil:false }");
+    shouldBeNonNull("gl = getWebGL(1, 1, {}, [ 0, 0, 0, 0 ], 1, 0)");
+    shouldBeTrue("gl.getContextAttributes().antialias");
+    shouldBeTrue("gl.getContextAttributes().depth");
+    shouldBeTrue("gl.getParameter(gl.DEPTH_BITS) >= 16");
+    shouldBeFalse("gl.getContextAttributes().stencil");
+    shouldBeTrue("gl.getParameter(gl.STENCIL_BITS) == 0");
+}
+
+function testDepth(depth)
+{
+    debug("");
+    debug("Testing depth = " + depth);
+    if (depth) {
+        shouldBeNonNull("gl = getWebGL(1, 1, { stencil: false, antialias: false }, [ 0, 0, 0, 1 ], 1, 0)");
+        shouldBeTrue("gl.getParameter(gl.DEPTH_BITS) >= 16");
+    } else {
+        shouldBeNonNull("gl = getWebGL(1, 1, { depth: false, stencil: false, antialias: false }, [ 0, 0, 0, 1 ], 1, 0)");
+        shouldBeTrue("gl.getParameter(gl.DEPTH_BITS) == 0");
+    }
+    shouldBeTrue("gl.getParameter(gl.RED_BITS) >= 8");
+    shouldBeTrue("gl.getParameter(gl.GREEN_BITS) >= 8");
+    shouldBeTrue("gl.getParameter(gl.BLUE_BITS) >= 8");
+    shouldBeTrue("gl.getParameter(gl.ALPHA_BITS) >= 8");
+
+    shouldBeNonNull("contextAttribs = gl.getContextAttributes()");
+    if (depth != contextAttribs.depth) {
+        testFailed("WebGL implementation doesn't obey depth attribute");
+    } else {
+        testPassed("WebGL implementation obeys depth attribute");
+    }
+
+    gl.depthFunc(gl.NEVER);
+
+    var vertices = new Float32Array([
+         1.0,  1.0, 0.0,
+        -1.0,  1.0, 0.0,
+        -1.0, -1.0, 0.0,
+         1.0,  1.0, 0.0,
+        -1.0, -1.0, 0.0,
+         1.0, -1.0, 0.0]);
+    var colors = new Uint8Array([
+        255, 0, 0, 255,
+        255, 0, 0, 255,
+        255, 0, 0, 255,
+        255, 0, 0, 255,
+        255, 0, 0, 255,
+        255, 0, 0, 255]);
+
+    draw(gl, vertices, colors);
+    var correctColor = contextAttribs.depth ? [0, 0, 0, 255] : [255, 0, 0, 255];
+    wtu.checkCanvasRect(gl, 0, 0, 1, 1, correctColor);
+}
+
+function testStencilAndDepth(stencil, depth)
+{
+    debug("");
+    debug("Testing stencil = " + stencil + ", depth = " + depth);
+    var creationString =
+        "gl = getWebGL(1, 1, { depth: " + depth + ", stencil: " + stencil + ", antialias: false }, [ 0, 0, 0, 1 ], 1, 0)";
+    shouldBeNonNull(creationString);
```

Comment:
Eval-ing as a side-effect is pretty gross. Can we be a bit more conservative about this?